presto-record-decoder/src/main/java/io/prestosql/decoder/json/AbstractDateTimeJsonValueProvider.java

```diff
@@ -81,7 +83,8 @@ public final long getLong()
             return packDateTimeWithZone(millis, getTimeZone());
         }
         if (type.equals(TIME_WITH_TIME_ZONE)) {
-            return packTimeWithTimeZone(millis * NANOSECONDS_PER_MILLISECOND, 0);
+            int offsetMinutes = getTimeZone().getZoneId().getRules().getOffset(Instant.ofEpochMilli(millis)).getTotalSeconds() / 60;
+            return packTimeWithTimeZone((millis + (offsetMinutes * 60 * MILLISECONDS_PER_SECOND)) * NANOSECONDS_PER_MILLISECOND, offsetMinutes);
```

Comment:
I am concerned by the fact that no tests were broken by those change. 
It looks like we are lacking decoding to `TIME WTZ` with input having timezone in input different than UTC.
Can you add test coverage to:
* `TestCustomDateTimeJsonFieldDecoder`
* `TestISO8601JsonFieldDecoder`
* `TestRFC2822JsonFieldDecoder` (optionally - as we should drop support for `TIME/TIME WTZ`  here)